libcflib/symbol_inspection.py

```diff
@@ -0,0 +1,51 @@
+import jedi
+
+import os
+
+from tqdm import tqdm
+
+
+def file_path_to_import(file_path: str):
+    return file_path.replace("/__init__.py", "").replace(".py", "").replace("/", ".")
+
+
+def get_all_symbol_names(top_dir):
+    # Note Jedi seems to pick up things that are protected by a
+    # __name__ == '__main__' if statement
+    # this could cause some over-reporting of viable imports this
+    # shouldn't cause issues with an audit since we don't expect 3rd parties
+    # to depend on those
+    symbols_dict = {}
+    module_import = top_dir.split("/")[-1]
+    # walk all the files looking for python files
+    for root, dirs, files in tqdm(os.walk(top_dir)):
+        _files = [f for f in files if f.endswith(".py")]
+        for file in _files:
```

Comment:
you might only want to walk _files if there is a `__init__.py`

Reply:
Yeah, those non-directory pkgs has burned me in the past.